src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -452,22 +449,45 @@ private async Task RunStepAsync(IStep step)
                 SetStepConclusion(step, Common.Util.TaskResultUtil.MergeTaskResults(step.ExecutionContext.Result, step.ExecutionContext.CommandResult.Value));
             }
 
+            if (step.ExecutionContext.Result == TaskResult.Failed)
+            {
+                var continueOnError = false;
+                try
+                {
+                    var templateEvaluator = step.ExecutionContext.ToPipelineTemplateEvaluator();
+                    continueOnError = templateEvaluator.EvaluateStepContinueOnError(step.ContinueOnError, step.ExecutionContext.ExpressionValues, step.ExecutionContext.ExpressionFunctions);
+                }
+                catch (Exception ex)
+                {
+                    Trace.Info("The step failed and an error occurred when attempting to determine whether to continue on error.");
+                    Trace.Error(ex);
+                    step.ExecutionContext.Error("The step failed and an error occurred when attempting to determine whether to continue on error.");
+                    step.ExecutionContext.Error(ex);
+                }
+
+                if (continueOnError)
+                {
+                    SetStepConclusion(step, TaskResult.Succeeded);
+                    Trace.Info($"Updated step result (continue on error)");
+                }
+            }
+
             Trace.Info($"Step result: {step.ExecutionContext.Result}");
             step.ExecutionContext.Debug($"Finished: {step.DisplayName}");
             step.ExecutionContext.PublishStepTelemetry();
         }
 
         private void SetStepConclusion(IStep step, TaskResult result)
         {
+            step.ExecutionContext.Outcome ??= result;
```

Comment:
Previously, we were setting the outcome as the result, since that was the only outcome. What this does is essentially:
`step.ExecutionContext.Outcome = step.ExecutionContext.Outcome ?? result`, or in other words, if we did not set the outcome, we set it to the result. Otherwise, favor the outcome set first since it is the worst possible scenario (if failure is the outcome previously, don't set it to success if continue-on-error sets the result to success)